
# HG changeset patch
# User Magnus Melin <mkmelin+mozilla@iki.fi>
# Date 1764969275 25200
# Node ID 107dcb12b088c68d909d69703c4cdfee87f5ff6d
# Parent  0d9d482fa04c32d7cc40c751df27e0cad2da4f73
Bug 2004103 - .calendar-displayed input title wording meaning was inversed. r=aleca


The logic was correct. It's just that the strings themselves are backwards! (show-calendar-title vs hide-calendar-title)

Differential Revision: https://phabricator.services.mozilla.com/D275185

diff --git a/calendar/base/content/calendar-management.js b/calendar/base/content/calendar-management.js
--- a/calendar/base/content/calendar-management.js
+++ b/calendar/base/content/calendar-management.js
@@ -139,16 +139,19 @@ async function loadCalendarManager() {
   } else {
     initHomeCalendar();
   }
 
   for (const calendar of sortCalendarArray(cal.manager.getCalendars())) {
     addCalendarItem(calendar);
   }
 
+  /**
+   * @param {calICalendar} calendar - Calendar to add to the calendar-list.
+   */
   function addCalendarItem(calendar) {
     const item = document
       .getElementById("calendar-list-item")
       .content.firstElementChild.cloneNode(true);
     const forceDisabled = calendar.getProperty("force-disabled");
     item.id = `calendar-listitem-${calendar.id}`;
     item.searchLabel = calendar.name;
     item.setAttribute("aria-label", calendar.name);
@@ -175,33 +178,34 @@ async function loadCalendarManager() {
     const colorMarker = item.querySelector(".calendar-color");
     if (calendar.getProperty("disabled")) {
       colorMarker.style.backgroundColor = "transparent";
       colorMarker.style.border = `2px solid var(--calendar-${cssSafeId}-backcolor)`;
     } else {
       colorMarker.style.backgroundColor = `var(--calendar-${cssSafeId}-backcolor)`;
     }
 
-    const label = item.querySelector(".calendar-name");
-    label.textContent = calendar.name;
+    item.querySelector(".calendar-name").textContent = calendar.name;
 
     updateCalendarStatusIndicators(item);
 
-    const enable = item.querySelector(".calendar-enable-button");
-    document.l10n.setAttributes(enable, "calendar-enable-button");
-
-    enable.hidden = forceDisabled || !calendar.getProperty("disabled");
+    const enableButton = item.querySelector(".calendar-enable-button");
+    document.l10n.setAttributes(enableButton, "calendar-enable-button");
+    enableButton.hidden = forceDisabled || !calendar.getProperty("disabled");
 
     const displayedCheckbox = item.querySelector(".calendar-displayed");
     displayedCheckbox.checked = calendar.getProperty("calendar-main-in-composite");
     displayedCheckbox.hidden = calendar.getProperty("disabled");
-    const stringName = cal.view.getCompositeCalendar(window).getCalendarById(calendar.id)
-      ? "hide-calendar-title"
-      : "show-calendar-title";
-    document.l10n.setAttributes(displayedCheckbox, stringName, { name: calendar.name });
+    document.l10n.setAttributes(
+      displayedCheckbox,
+      displayedCheckbox.checked ? "show-calendar-title" : "hide-calendar-title",
+      {
+        name: calendar.name,
+      }
+    );
 
     calendarList.appendChild(item);
     if (calendar.getProperty("calendar-main-default")) {
       // The list needs to handle the addition of the row before we can select it.
       setTimeout(() => {
         calendarList.selectedIndex = calendarList.rows.indexOf(item);
       });
     }
@@ -230,24 +234,29 @@ async function loadCalendarManager() {
     if (!event.target.matches(".calendar-displayed")) {
       return;
     }
 
     const item = event.target.closest("li");
     const calendarId = item.getAttribute("calendar-id");
     const calendar = cal.manager.getCalendarById(calendarId);
 
-    if (event.target.checked) {
+    const displayedCheckbox = event.target;
+    if (displayedCheckbox.checked) {
       compositeCalendar.addCalendar(calendar);
     } else {
       compositeCalendar.removeCalendar(calendar);
     }
-
-    const stringName = event.target.checked ? "hide-calendar-title" : "show-calendar-title";
-    document.l10n.setAttributes(event.target, stringName, { name: calendar.name });
+    document.l10n.setAttributes(
+      displayedCheckbox,
+      displayedCheckbox.checked ? "show-calendar-title" : "hide-calendar-title",
+      {
+        name: calendar.name,
+      }
+    );
 
     calendarList.focus();
   });
   calendarList.addEventListener("dblclick", event => {
     if (
       event.target.matches(".calendar-displayed") ||
       event.target.matches(".calendar-enable-button")
     ) {
@@ -274,27 +283,29 @@ async function loadCalendarManager() {
     const calendarId = item.getAttribute("calendar-id");
     const calendar = cal.manager.getCalendarById(calendarId);
 
     switch (event.key) {
       case "Delete":
         promptDeleteCalendar(calendar);
         break;
       case " ": {
-        if (item.querySelector(".calendar-displayed").checked) {
+        const displayedCheckbox = item.querySelector(".calendar-displayed");
+        if (displayedCheckbox.checked) {
           compositeCalendar.removeCalendar(calendar);
         } else {
           compositeCalendar.addCalendar(calendar);
         }
-        const stringName = item.querySelector(".calendar-displayed").checked
-          ? "hide-calendar-title"
-          : "show-calendar-title";
-        document.l10n.setAttributes(item.querySelector(".calendar-displayed"), stringName, {
-          name: calendar.name,
-        });
+        document.l10n.setAttributes(
+          displayedCheckbox,
+          displayedCheckbox.checked ? "show-calendar-title" : "hide-calendar-title",
+          {
+            name: calendar.name,
+          }
+        );
         break;
       }
     }
   });
   calendarList.addEventListener("select", () => {
     const item = calendarList.rows[calendarList.selectedIndex];
     const calendarId = item.getAttribute("calendar-id");
     const calendar = cal.manager.getCalendarById(calendarId);
@@ -434,17 +445,16 @@ function unloadCalendarManager() {
   cal.manager.removeObserver(calendarList._calendarManagerObserver);
 }
 
 /**
  * A handler called to set up the context menu on the calendar list.
  *
  * @param {Event} event - The click DOMEvent.
  */
-
 function calendarListSetupContextMenu(event) {
   let calendar;
   const composite = cal.view.getCompositeCalendar(window);
 
   if (event.target.matches(".calendar-displayed")) {
     return;
   }
 

